VideosController.js

```diff
@@ -0,0 +1,44 @@
+
+var express = require('express');
+var router = express.Router();
+var Events = require('./Events');
+
+router.get('/', function (req, res) {
+    Events.find({
+        $or: [{event: "StreamCreated"}, {event: "StreamEnded"}]
+    }).sort({
+        createdAt: 1
+    }).exec(function (err, events) {
+        if (err) return res.status(500).send("There was a problem finding the video.")
+
+        videos = {};
+        for (let i in events) {
+            event = events[i]
+            switch (event.event) {
+                case "StreamCreated": 
+                    idx = event.properties["manifestID"] + event.properties["createTime"]
+                    videos[idx] = {manifestID: event.properties["manifestID"], createTime: event.properties["createTime"], success: false}
+                    break;
+                case "StreamEnded": 
+                    idx = event.properties["manifestID"] + event.properties["createTime"]
+                    video = videos[idx]
+                    if (video != null) {
+                        //set video end time
+                        video["endTime"] = event.properties["endTime"]
+                        //set status to succes
+                        video["success"] = true
+                    } else {
+                        //something weird has happened
+                    }
+                    break
+                case "TranscodeSuccess": 
+                    //Add to the video's length
+                    break
+            }
+        }
+
+        res.status(200).send(videos)
```

Comment:
Probably because I don't know the details of Mongo-JS interop, but how does the `video` object actually get stored in the db?

Reply:
Ah, okay, now looking over the code, let me know if this flow is correct:

1. Livepeer node submits events to the `EventsController` , which gets stored in Mongo
2. After some period of time, another process queries Mongo and retrieves a batch of events. (I'm guessing this part is unimplemented?)
3. The batch of events gets submitted to `VideosController`, which reconstructs the state of each stream/manifest from there (in the `videos` object).
4. The state gets returned to the caller

Reply:
Naa, I get it now: step 2 is actually the `Events.Find` call which queries Mongo. For some reason I thought that was a filter which acted on events submitted as part of the HTTP request. Okay, makes sense now!